apps/web/src/components/Day.tsx

```diff
@@ -116,6 +118,29 @@ export const DayContent = (props: DayContentProps) => {
             newTasksOrder,
           },
         },
+        optimisticUpdater: (store) => {
+          // get the previous date of the task
+          const {from, movedTaskId, htmlParent  } = updateTaskDateInfo;
+          const fromLocation = store.get(`Day_${from.id}`);
+          //get all tasks for the previous data
+          const fromTasks = fromLocation?.getLinkedRecords("tasks");
+          // remove the task from the current date
+          const newFromTasks = fromTasks?.filter(
+            (t) => t.getDataID() !== updateTaskDateInfo.movedTaskId
+          );
+          fromLocation?.setLinkedRecords(newFromTasks, "tasks");
+          //get the task you are moving
+          const taskToMove = fromTasks?.find(
+            (t) => t.getDataID() === movedTaskId
+          );
+          //get the date you ingtend to move it to
+          const to = store.get(`Day_${htmlParent.id}`);
+          //get the tasks that are already under the intended date
+          const toTasks = to?.getLinkedRecords("tasks");
+          // move the task to the new date
+          const newToTasks = !!taskToMove ? [taskToMove, ...(toTasks ?? [])] : toTasks;
```

Comment:
This seems to always put the task at the top of the day, even if the user puts it at the bottom or somewhere in the middle. Can it be changed to also take into account the order?

Reply:
We can try and get it to work like that. However, it still behaves the same way (always putting the task to the bottom) even without the optimistic update.

Reply:
Currently investigating the task disappearance.

Reply:
It's ok if you don't find the reason. I will experiment with your code and see if it can be done. If all fails, I will go back using ReactSortable's normal behavior and revert [this change](https://github.com/richardguerre/flow/pull/3/files#diff-c81ba4292668c8be30e9d6ed4561028bbb478a980329a75aacaa4c2f00b1ed3cL130-R155)

Reply:
I found the reason this is not working properly and sometimes the task disappears. It's flakey, but this is the current order of actions:
1. User moves task to a different day or location.
2. handleTaskMove is triggered and updates the `updateTaskDateInfo` state  (this state is required to force React to re-render and wait for the task to be in the DOM).
3. The task is moved back into its original day as there was no changes to the `tasks` state and `ReactSortable` renders exactly what is in the `task` state.
4. The useEffect that detects changes in the `updateTaskDateInfo` state, uses `htmlParent` (i.e. `e.to`) to get the day and task IDs where the user tried to move the task, but the task is not in there as ReactSortable moved it back into the original day.
5. The mutation is triggered using that day and its tasks, without the task that the user tried to move in the `newTasksOrder` input.
6. Relay's optimistic update occurs, which updates the store and updates the `tasks` state (on [these line](https://github.com/richardguerre/flow/pull/3/files#diff-c81ba4292668c8be30e9d6ed4561028bbb478a980329a75aacaa4c2f00b1ed3cR103-R105)). This update adds the task to the new day.
7. The backend resolves the mutation and returns both days missing the task as both days' tasksOrder in the backend are missing the task ID that was moved.

The above is what I gathered after an hour of experimenting on it and it was a bit different several times.

Considering this, I would go back to using `setList={setTasks}` or simply closing this PR. I do however have a need for a `optimisticUpdater` for this `updateTaskDate` mutation where if the task was originally `DONE` but then moved to the future, the backend changes its status to `TODO`, and vice-versa for a task in `TODO` but moving into the past. I'll create a new ticket for that so that there is a new estimate for it.